### DISCUSSION TITLE
invalid ELF header (Amazon Lambda related?)

### DISCUSSION THREAD

**Discussion by fsobh** [samizdatco/skia-canvas#76] [Q&A] [answered]
undefined	ERROR	Uncaught Exception 	
{
    "errorType": "Error",
    "errorMessage": "/var/task/node_modules/skia-canvas/lib/v6/index.node: invalid ELF header",
    "code": "ERR_DLOPEN_FAILED",
    "stack": [
        "Error: /var/task/node_modules/skia-canvas/lib/v6/index.node: invalid ELF header",
        "    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1144:18)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
        "    at Module.require (internal/modules/cjs/loader.js:974:19)",
        "    at require (internal/modules/cjs/helpers.js:93:18)",
        "    at Object.<anonymous> (/var/task/node_modules/skia-canvas/lib/index.js:23:29)",
        "    at Module._compile (internal/modules/cjs/loader.js:1085:14)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)"
    ]
}

**Comment by meihuanyu** [comment]
can‘t move node_modules to different platform。

**Comment by samizdatco** [comment, marked as answer]
That stack trace doesn't give me much to go on, but googling for your error message makes it seem like you're trying to run this on Amazon Lambda. Based on this [stack overflow answer](https://stackoverflow.com/questions/29994411/invalid-elf-header-when-using-the-nodejs-ref-module-on-aws-lambda) it sounds like you may have tried to run the module that was compiled for your dev machine on the server. 

Try using docker as the answer suggests. Something like this:

```console
$ mkdir my-project
$ cd my-project
$ npm init -y
$ docker run --rm -v "$PWD":/var/task lambci/lambda:build-nodejs12.x npm i skia-canvas
```

The `node_modules` directory that it sets up should compatible with Amazon Linux.
